src/create.rs

```diff
@@ -95,14 +101,18 @@ fn run_container<P: AsRef<Path>>(
     rootfs: PathBuf,
     spec: oci_spec::Spec,
     csocketfd: Option<FileDescriptor>,
+    systemd_cgroup: bool,
     container: Container,
     command: impl Command,
 ) -> Result<Process> {
     prctl::set_dumpable(false).unwrap();
     let linux = spec.linux.as_ref().unwrap();
     let namespaces: Namespaces = linux.namespaces.clone().into();
 
-    let cmanager = cgroups::Manager::new(linux.cgroups_path.clone())?;
+    let cmanager: Box<dyn cgroups::Manager> = match systemd_cgroup {
```

Comment:
@utam0k  I think I can use a `&dyn cgroups::Manager` by [allocating a variable for both managers](https://stackoverflow.com/a/35261774/3891733), but I need to check (thus dropping the heap allocation).

Reply:
I think this implementation is good.  Have you come up with any other good ways to implement this?

Reply:
I need to experiment a little more but this seems like the general direction. We could also avoid a `trait object` here and have `Manager` be an `enum` with `Manager::CgroupsFS(struct)` and `Manager::SystemdD(struct)` variants.

I think both are valid ways, this seems nicer.. Not sure which is more compatible with "The rust way"..

Reply:
I also think it's better to use Trait, so we'll adopt that! Thank you very much.